b2sdk/transfer/inbound/downloaded_file.py

```diff
@@ -203,3 +203,24 @@ def save_to(self, path_, mode='wb+', allow_seeking=True):
             buffering=self.write_buffer_size,
         ) as file:
             self.save(file, allow_seeking=allow_seeking)
+
+    def safe_save_to(self, path_, mode='wb+', allow_seeking=True):
+        """
+        Provides a safe way to save data to a file.
+
+        This method acts as a wrapper around `save_to`. It first attempts to save
+        data in the provided mode. If the mode is 'wb+' and an unsupported
+        operation error arises due to seeking issues, it falls back to 'wb' mode
+        and retries the operation.
+
+        :param path_: path to file to be opened
+        :param mode: mode in which the file should be opened
+        :param allow_seeking: if False, download strategies that rely on seeking to write data
+                              (parallel strategies) will be discarded.
+        """
+        try:
+            self.save_to(path_, mode, allow_seeking)
+        except io.UnsupportedOperation:
+            if mode != 'wb+':
+                raise
+            self.save_to(path_, 'wb', allow_seeking)
```

Comment:
pretty sure this is not going to work depending on the download strategy, if not for all of them with sufficiently large file

also why new method if this is `safe`?

Reply:
- Of course, I'm not sure this is working for all cases. So I tried to test this code for several cases such as `Parallel` .
And also tested on `multi-thread` mode and confirmed everything is okay. Because of this, I changed the `allow_seeking`.
- And regardig the why I created the `safe` mode, beceause this issues is arising for non-seekable files.
I think try to download files to non-seekable file such as `/dev/stdout` is one of rarly cases. and it also affect to the performance and I think we can use original `save_to` function for eager running mode.

Reply:
current implementation may cause issues:
* we have new method, which IMO doesn't quite make sense, if we can support all in one - this will just confuse users and give more work for maintainers
* if UnsupportedOperation (Which is quite vague) is raised but anything other than `open` it will cause very nasty and hard to detect bugs, since part of the download stream will be consumed by first try

better to figure out why we need `wb+` - I'm not sure if we need it at all, or will `+` suffice.
Maybe if allow_seeking is True requires `wb+`, but otherwise, we should do without it IMO.